### PR TITLE
:running: Release artifacts and conformance test tweaks

### LINKED ISSUE #1500: Support clusterctl by implementing provider contract
/kind feature

**Describe the solution you'd like**
After following instructions [in the quickstart](https://master.cluster-api.sigs.k8s.io/clusterctl/overview.html)
I get the following error when running `clusterctl config cluster my-cluster`
```
$ clusterctl config cluster my-cluster --infrastructure=aws:v0.5.0
Error: failed to read "config-kubeadm.yaml" from provider's repository "aws": failed to get GitHub release v0.5.0: failed to read release "v0.5.0": GET https://api.github.com/repos/kubernetes-sigs/cluster-api-provider-aws/releases/tags/v0.5.0: 404 Not Found []
```
I'm using `clusterctl-settings.json`.
```
{
      "providers": [ "cluster-api", "kubeadm-bootstrap", "aws"],
      "provider_repos": ["../cluster-api-provider-aws"]
}
```

We need to follow the [provider contract and provide the workload cluster template](https://master.cluster-api.sigs.k8s.io/clusterctl/provider-contract.html#workload-cluster-templates) `config-kubeadm.yaml`

**Environment:**
- Cluster-api version: https://github.com/kubernetes-sigs/cluster-api/tree/7816c0c01a97824103a2ee8fa8078625f0d56d36
- Cluster-api-provider-aws version: 00d21035bb4d0ea3fd49a77ffd5811529abeaa3b

### LINKED ISSUE #1605: Examples should ship with KCP for v1alpha3
/kind feature

**Describe the solution you'd like**
The examples directory contains manifests users can use to spin up a new cluster. Instead of using simple machines, we should rely on KCP functionality in https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/examples/controlplane/controlplane.yaml

### PR DESCRIPTION
**Pull request by detiber**
**What this PR does / why we need it**:
Updates to move towards templates vs the old examples and tweaks to make the conformance tests happy with the change

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1605 
Fixes #1500

### CODE REVIEW

File: templates/kustomizeversions.yaml [hunk 1]
```diff
@@ -0,0 +1,166 @@
+apiVersion: controlplane.cluster.x-k8s.io/v1alpha3
```
Comment (randomvariable) [inline templates/kustomizeversions.yaml]: Should have a follow up issue to move conformance stuff moved into a separate directory so it doesn't confuse consumers.

Comment (randomvariable) [reply templates/kustomizeversions.yaml]: https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/1610

File: templates/cluster-template.yaml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+kind: AWSMachineTemplate
+metadata:
+  name: "${CLUSTER_NAME}-md-0"
+spec:
+  template:
+    spec:
+      instanceType: "${NODE_MACHINE_TYPE}"
+      iamInstanceProfile: "nodes.cluster-api-provider-aws.sigs.k8s.io"
+      sshKeyName: "${SSH_KEY_NAME}"
```
Comment (randomvariable) [inline templates/cluster-template.yaml]:
```suggestion
      sshKeyName: "${AWS_SSH_KEY_NAME}"
```

File: templates/cluster-template.yaml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    apiVersion: controlplane.cluster.x-k8s.io/v1alpha3
+    name: "${CLUSTER_NAME}-control-plane"
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+kind: AWSCluster
+metadata:
+  name: "${CLUSTER_NAME}"
+spec:
+  region: "${AWS_REGION}"
+  sshKeyName: "${SSH_KEY_NAME}"
```
Comment (randomvariable) [inline templates/cluster-template.yaml]:
```suggestion
  sshKeyName: "${AWS_SSH_KEY_NAME}"
```

File: templates/cluster-template.yaml [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+kind: AWSMachineTemplate
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+metadata:
+  name: "${CLUSTER_NAME}-control-plane"
+spec:
+  template:
+    spec:
+      instanceType: "${CONTROL_PLANE_MACHINE_TYPE}"
+      iamInstanceProfile: "control-plane.cluster-api-provider-aws.sigs.k8s.io"
+      sshKeyName: "${SSH_KEY_NAME}"
```
Comment (randomvariable) [inline templates/cluster-template.yaml]:
```suggestion
      sshKeyName: "${AWS_SSH_KEY_NAME}"
```

File: hack/ci/e2e-conformance.sh [hunk 5]
```diff
@@ -23,7 +23,6 @@ AWS_REGION=${AWS_REGION:-"us-east-1"}
 CLUSTER_NAME=${CLUSTER_NAME:-"test-$(date +%s)"}
 SSH_KEY_NAME=${SSH_KEY_NAME:-"${CLUSTER_NAME}-key"}
```
Comment (randomvariable) [inline hack/ci/e2e-conformance.sh]:
```suggestion
AWS_SSH_KEY_NAME=${AWS_SSH_KEY_NAME:-"${CLUSTER_NAME}-key"}
```

File: hack/ci/e2e-conformance.sh [hunk 6]
```diff
@@ -100,22 +99,20 @@ function ssh-to-node() {
 
   ssh_key_pem="/tmp/${SSH_KEY_NAME}.pem"
```
Comment (randomvariable) [inline hack/ci/e2e-conformance.sh]:
```suggestion
  ssh_key_pem="/tmp/${AWS_SSH_KEY_NAME}.pem"
```

File: hack/ci/e2e-conformance.sh [hunk 7]
```diff
@@ -275,25 +263,25 @@ fix_manifests() {
...
-  sed -i 's|CI_VERSION=.*|CI_VERSION='$CI_VERSION'|' examples/_out/machinedeployment.yaml
+  sed -i 's|kubernetesVersion: .*|kubernetesVersion: "ci/'"${CI_VERSION}"'"|' examples/_out/controlplane.yaml
+  sed -i 's|CI_VERSION=.*|CI_VERSION='"$CI_VERSION"'|' examples/_out/controlplane.yaml
+  sed -i 's|CI_VERSION=.*|CI_VERSION='"$CI_VERSION"'|' examples/_out/machinedeployment.yaml
 }
 
 
 create_key_pair() {
-  (aws ec2 create-key-pair --key-name ${SSH_KEY_NAME} --region ${AWS_REGION} > /tmp/keypair-${SSH_KEY_NAME}.json \
+  (aws ec2 create-key-pair --key-name "${SSH_KEY_NAME}" --region "${AWS_REGION}" > /tmp/keypair-"${SSH_KEY_NAME}".json \
```
Comment (randomvariable) [inline hack/ci/e2e-conformance.sh]:
```suggestion
  (aws ec2 create-key-pair --key-name "${AWS_SSH_KEY_NAME}" --region "${AWS_REGION}" > /tmp/keypair-"${AWS_SSH_KEY_NAME}".json \
```

File: hack/ci/e2e-conformance.sh [hunk 8]
```diff
@@ -275,25 +263,25 @@ fix_manifests() {
...
    && KEY_PAIR_CREATED="true" \
-   && jq -r '.KeyMaterial' /tmp/keypair-${SSH_KEY_NAME}.json > /tmp/${SSH_KEY_NAME}.pem \
-   && chmod 600 /tmp/${SSH_KEY_NAME}.pem) || true
+   && jq -r '.KeyMaterial' /tmp/keypair-"${SSH_KEY_NAME}".json > /tmp/"${SSH_KEY_NAME}".pem \
+   && chmod 600 /tmp/"${SSH_KEY_NAME}".pem) || true
```
Comment (randomvariable) [inline hack/ci/e2e-conformance.sh]:
```suggestion
   && chmod 600 /tmp/"${AWS_SSH_KEY_NAME}".pem) || true
```

File: hack/ci/e2e-conformance.sh [hunk 9]
```diff
@@ -275,25 +263,25 @@ fix_manifests() {
...
 }
 
 delete_key_pair() {
   # Delete only if we created it
   if [[ "${KEY_PAIR_CREATED:-}" = true ]]; then
-    aws ec2 delete-key-pair --key-name ${SSH_KEY_NAME} --region ${AWS_REGION} || true
-    rm /tmp/keypair-${SSH_KEY_NAME}.json || true
-    rm /tmp/${SSH_KEY_NAME}.pem || true
+    aws ec2 delete-key-pair --key-name "${SSH_KEY_NAME}" --region "${AWS_REGION}" || true
+    rm /tmp/keypair-"${SSH_KEY_NAME}".json || true
```
Comment (randomvariable) [inline hack/ci/e2e-conformance.sh]:
```suggestion
    rm /tmp/keypair-"${AWS_SSH_KEY_NAME}".json || true
```

File: hack/ci/e2e-conformance.sh [hunk 10]
```diff
@@ -275,25 +263,25 @@ fix_manifests() {
...
+    rm /tmp/"${SSH_KEY_NAME}".pem || true
```
Comment (randomvariable) [inline hack/ci/e2e-conformance.sh]:
```suggestion
    rm /tmp/"${AWS_SSH_KEY_NAME}".pem || true
```

File: hack/ci/e2e-conformance.sh [hunk 11]
```diff
@@ -302,21 +290,36 @@ create_cluster() {
...
+  set +o xtrace
+
   # Load the newly built image into kind and start the cluster
+  AWS_REGION=${AWS_REGION} \
+  CONTROL_PLANE_MACHINE_COUNT=1 \
+  WORKER_MACHINE_COUNT=2 \
+  KUBERNETES_VERSION=$KUBERNETES_VERSION \
+  IMAGE_ID=$image_id \
+  CLUSTER_NAME=$CLUSTER_NAME \
+  SSH_KEY_NAME=$SSH_KEY_NAME \
```
Comment (randomvariable) [inline hack/ci/e2e-conformance.sh]:
```suggestion
  AWS_SSH_KEY_NAME=$AWS_SSH_KEY_NAME \
```

**Comment by detiber** [comment]
@chuckha @randomvariable I believe this is ready for review now...  The e2e tests are currently running and the conformance tests that I triggered on a previous push (where the only changes made since were in aws_test.go) is also still running. Previously both were failing during test setup.

**Comment by detiber** [comment]
Currently running conformance test: https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-aws/1607/pull-cluster-api-provider-aws-make-conformance/1234555360723865601

**Pull request merged**